File: ch04/AlertView/AlertView/ViewController.m
//
//  ViewController.m
//  AlertView
//
//  Created by MyZiyue on 2018/12/12.
//  Copyright © 2018 MyZiyue.com. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    CGRect screen = [[UIScreen mainScreen] bounds];
    
    // 1. AlertView
    UIButton* buttonAlertView = [UIButton buttonWithType:UIButtonTypeSystem];
    [buttonAlertView setTitle:@"Test警告框" forState: UIControlStateNormal];
    
    CGFloat buttonAlertViewWidth = 100;
    CGFloat buttonAlertViewHeight = 30;
    CGFloat buttonAlertViewTopView = 30;
    
    buttonAlertView.frame = CGRectMake((screen.size.width - buttonAlertViewWidth)/2, buttonAlertViewTopView, buttonAlertViewWidth, buttonAlertViewHeight);
    // 指定事件处理方法
    [buttonAlertView addTarget:self action:@selector(testAlertView:) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:buttonAlertView];
    
    // 2. AlertViewSheet
    UIButton* buttonActionSheet = [UIButton buttonWithType:UIButtonTypeSystem];
    [buttonActionSheet setTitle:@"Test操作表" forState:UIControlStateNormal];
    
    CGFloat buttonActionSheetWidth = 100;
    CGFloat buttonActionSheetHeight = 30;
    CGFloat buttonActionSheetTopView = 260;
    
    buttonActionSheet.frame = CGRectMake((screen.size.width - buttonActionSheetWidth)/2, buttonActionSheetTopView, buttonActionSheetWidth, buttonActionSheetHeight);
    // 指定事件处理方法
    [buttonActionSheet addTarget:self action:@selector(testActionSheet:) forControlEvents:UIControlEventTouchUpInside];;
    [self.view addSubview:buttonActionSheet];
}

-(void)testAlertView:(id)sender {
    UIAlertController* alertController = [UIAlertController alertControllerWithTitle:@"Alert" message:@"Alert text goes here" preferredStyle:UIAlertControllerStyleAlert];
    
    UIAlertAction* noAction = [UIAlertAction actionWithTitle:@"No" style:UIAlertActionStyleCancel handler:^(UIAlertAction * _Nonnull action) {
        NSLog(@"Tap No Button");
    }];
    
    UIAlertAction* yesAction = [UIAlertAction actionWithTitle:@"Yes" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        NSLog(@"Tap Yes Button");
    }];
    
    [alertController addAction:noAction];
    [alertController addAction:yesAction];
    
    // 显示
    [self presentViewController:alertController animated:true completion: nil];
}

-(void)testActionSheet:(id)sender {
    UIAlertController* alertControllerSheet = [[UIAlertController alloc] init];
    
    UIAlertAction* cancelAction = [UIAlertAction actionWithTitle:@"Cancel" style:UIAlertActionStyleCancel handler:^(UIAlertAction * _Nonnull action) {
        NSLog(@"Tap Cancel Button");
    }];
    UIAlertAction* destructionAction = [UIAlertAction actionWithTitle:@"Destruction" style:UIAlertActionStyleDestructive handler:^(UIAlertAction * _Nonnull action) {
        NSLog(@"Tap Destruction Button");
    }];
    UIAlertAction* otherAction = [UIAlertAction actionWithTitle:@"Others" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        NSLog(@"Tap Others Button");
    }];
    
    [alertControllerSheet addAction:cancelAction];
    [alertControllerSheet addAction:destructionAction];
    [alertControllerSheet addAction:otherAction];
    
    [self presentViewController:alertControllerSheet animated:true completion:nil];
}

@end

File: ch04/SAlertView/SAlertView/ViewController.swift
//
//  ViewController.swift
//  SAlertView
//
//  Created by MyZiyue on 2018/12/12.
//  Copyright © 2018 MyZiyue.com. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let screen = UIScreen.main.bounds
        
        // 1. AlertView
        let buttonAlertView = UIButton(type: UIButton.ButtonType.system)
        buttonAlertView.setTitle("Test警告框", for: UIControl.State.normal)
        
        let buttonAlertViewWidth: CGFloat = 100
        let buttonAlertViewHeight: CGFloat = 30
        let buttonAlertViewTopView: CGFloat = 130
        
        buttonAlertView.frame = CGRect(x:(screen.size.width - buttonAlertViewWidth)/2, y:buttonAlertViewTopView, width:buttonAlertViewWidth, height:buttonAlertViewHeight)
        // 指定事件处理方法
        buttonAlertView.addTarget(self, action: #selector(testAlertView), for: UIControl.Event.touchUpInside)
        self.view.addSubview(buttonAlertView)
        
        // 2. AlertViewSheet
        let buttonActionSheet = UIButton(type: UIButton.ButtonType.system)
        buttonActionSheet.setTitle("Test操作表", for: UIControl.State.normal)
        
        let buttonActionSheetWidth: CGFloat = 100
        let buttonActionSheetHeight: CGFloat = 30
        let buttonActionSheetTopView: CGFloat = 260
        
        buttonActionSheet.frame = CGRect(x:(screen.size.width - buttonActionSheetWidth)/2, y:buttonActionSheetTopView, width:buttonActionSheetWidth, height:buttonActionSheetHeight)
        // 指定事件处理方法
        buttonActionSheet.addTarget(self, action: #selector(testActionSheet), for: UIControl.Event.touchUpInside)
        self.view.addSubview(buttonActionSheet)
    }
    
    @objc func testAlertView(sender: AnyObject) {
        let alertController: UIAlertController = UIAlertController(title: "Alert", message: "Alert text goes here", preferredStyle: UIAlertController.Style.alert)
        
        let noAction = UIAlertAction(title: "No", style: .cancel) {
            (alertAction) -> Void in  NSLog("Tap No Button")
        }
        
        let yesAction = UIAlertAction(title: "Yes", style: .default) {
            (alertAction) -> Void in  NSLog("Tap Yes Button")
        }
        
        alertController.addAction(noAction)
        alertController.addAction(yesAction)
        
        // 显示
        self.present(alertController, animated: true, completion: nil)
    }
    
    @objc func testActionSheet(sender: AnyObject) {
        let actionSheetController: UIAlertController = UIAlertController()
        
        let cancelAction = UIAlertAction(title: "Cancel", style: UIAlertAction.Style.cancel) { (alertAction) -> Void in
            NSLog("Tap Cancel Button")
        }
        let destructiveAction = UIAlertAction(title: "破坏性按钮", style: UIAlertAction.Style.destructive) { (alertAction) -> Void in
            NSLog("Tap Destructive Button")
        }
        let otherAction = UIAlertAction(title: "其他按钮", style: UIAlertAction.Style.default) { (alertAction) -> Void in
            NSLog("Tap Others Button")
        }
        
        actionSheetController.addAction(cancelAction)
        actionSheetController.addAction(destructiveAction)
        actionSheetController.addAction(otherAction)
        
        self.present(actionSheetController, animated: true, completion: nil)
    }


}

